features/flows/classification/engine/impl/src/test/java/org/opennms/netmgt/flows/classification/internal/DefaultClassificationEngineTest.java

```diff
@@ -191,6 +192,18 @@ public void verifyAllPortsToEnsureEngineIsProperlyInitialized() {
         }
     }
 
+    // See NMS-12429
+    @Test
+    public void verifyDoesNotRunOutOfMemory() {
+        final List<Rule> rules = Lists.newArrayList();
+        for (int i=0; i<100; i++) {
+            final Rule rule = new RuleBuilder().withName("rule1").withPosition(i+1).withProtocol("UDP").withDstAddress("192.168.0." + i).build();
+            rules.add(rule);
+        }
+        final DefaultClassificationEngine engine = new DefaultClassificationEngine(() -> rules, FilterService.NOOP);
+        System.gc();
```

Comment:
What's the purpose of this explicit gc call?

Reply:
The next test is using a timeout and I saw it fail because too much memory was allocated and cleaning it took too long. The `System.gc()` call should prevent that.

Reply:
Can we add a comment explaining that?

Reply:
Yep. I forgot to push that.

Reply:
Turns out, the whole thing was damn slow instead of just using memory. Reworked it so no `System.gc()` call required.